machine.go

```
@@ -103,12 +104,47 @@ func UpdateComponentCloudAttributes(ctx context.Context, c *app.AppClient, id st
 	if err != nil {
 		return err
 	}
+	found, err := updateComponentOrServiceConfig(part.RobotConfig, name, newAttr)
+	if err != nil {
+		return err
+	}
 
-	cs, ok := part.RobotConfig["components"].([]interface{})
+	fragments, hasFragments := part.RobotConfig["fragments"].([]interface{})
+
+	// check fragments
+	if !found && hasFragments {
+		for _, frag := range fragments {
+			fID, version, err := checkFragmentInConfig(frag)
+			if err != nil {
+				return err
+			}
+			// first, determine which fragment has the component.
+			fragModString, err := findComponentInFragment(ctx, c.GetFragment, fID, version, name)
+			if err != nil {
+				continue
```

hate swallowing errors